client/signup/main.jsx

```diff
@@ -542,7 +542,9 @@ class Signup extends Component {
 		// to invalid step.
 		if ( stepName && ! this.isEveryStepSubmitted() ) {
 			const locale = ! this.props.isLoggedIn ? this.props.locale : '';
-			page( getStepUrl( flowName, stepName, stepSectionName, locale ) );
+			const queryObject = ( this.props.initialContext && this.props.initialContext.query ) || {};
+			const siteSlugOrId = pick( queryObject, [ 'siteSlug', 'siteId' ] );
```

Comment:
Not important but `siteIdOrSlug` might be better as we use this term everywhere in our codebase except one.